### PR TITLE
PT1 Stable Release Distributed Documentation

### PR DESCRIPTION
**Pull request by teng-li**
The doc covers pretty much all we have had on distributed for PT1 stable release, tracked in https://github.com/pytorch/pytorch/issues/14080

Tested by previewing the sphinx generated webpages. All look good.

### CODE REVIEW

File: docs/source/distributed.rst [hunk 1]
```diff
@@ -314,6 +405,25 @@ Launch utility
 --------------
 
 The `torch.distributed` package also provides a launch utility in
-`torch.distributed.launch`.
+`torch.distributed.launch`. This helper utility is helpful to launch
```
Comment (SsnL) [inline docs/source/distributed.rst]: too many "help"s. helpful -> useful, maybe?

File: docs/source/distributed.rst [hunk 2]
```diff
@@ -314,6 +405,25 @@ Launch utility
...
+
+Spawn utility
+-------------
+
+The :doc:`multiprocessing` package also provides a spawn utility in
+:func:`torch.multiprocessing.spawn`. This helper utility is helpful to spawn
+multiple processes per node by passing in the function that needs to run from
+multiple processes, and can be used for multiprocess distributed training as well.
+
+For references on how to use it, please refer to `PyToch example - imagenet
```
Comment (SsnL) [inline docs/source/distributed.rst]: ImageNet

File: docs/source/distributed.rst [hunk 3]
```diff
@@ -314,6 +405,25 @@ Launch utility
...
+
 
 .. automodule:: torch.distributed.launch
+
+
+Spawn utility
+-------------
+
+The :doc:`multiprocessing` package also provides a spawn utility in
+:func:`torch.multiprocessing.spawn`. This helper utility is helpful to spawn
```
Comment (SsnL) [inline docs/source/distributed.rst]: same here. maybe "useful"?

**Review by ssnl — APPROVED** [review]

File: docs/source/distributed.rst [hunk 4]
```diff
@@ -26,17 +29,98 @@ MPI supports cuda only if the implementation used to build PyTorch supports it.
...
+| barrier    | ✓   | ✘   | ✓   | ?   | ✘   | ✓   |
 +------------+-----+-----+-----+-----+-----+-----+
 
+
+Backends that comes with PyTorch
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+PyTorch distributed currently only supports Linux. By default, two backends: NCCL and Gloo
+will be built and included in PyTorch distributed, when CUDA is available. MPI is an
+optional backend that can only be included if you build PyTorch from source. (e.g.
```
[the remark this answers is not in the document]
Comment (teng-li) [reply docs/source/distributed.rst]: e.g. should be fine, that's what I meant initially

**Pull request closed without merging**